src/compiler/checker.ts

```diff
@@ -27574,6 +27307,23 @@ namespace ts {
                 : getDeclaredTypeOfSymbol(classSymbol);
         }
 
+        function getClassElementPropertyKeyType(element: ClassElement) {
+            const name = element.name!;
+            switch (name.kind) {
+                case SyntaxKind.Identifier:
+                    return getLiteralType(idText(name));
+                case SyntaxKind.NumericLiteral:
+                case SyntaxKind.StringLiteral:
+                    return getLiteralType(name.text);
+                case SyntaxKind.ComputedPropertyName:
+                    const nameType = checkComputedPropertyName(name);
+                    return isTypeAssignableToKind(nameType, TypeFlags.ESSymbolLike) ? nameType : stringType;

```

Comment:
This should probably be handling late bound names.
eg, 
```ts
const methodName = "method";
class Foo {
  @dec [methodName] {}
}
```
Unless I'm mistaken and for some reason checking with the more specific name is undesirable.